lambdas/mesh-poll/src/processor.py

```diff
@@ -0,0 +1,175 @@
+
+        self.__event_publisher = EventPublisher(
+            event_bus_arn=event_bus_arn,
+            dlq_url=dlq_url,
+            logger=self.__log
+        )
+
+    def is_enough_time_to_process_message(self):
+        """
+        Determines whether the lambda should continue to process messages
+        """
+        remaining_time_in_millis = self.__get_remaining_time_in_millis()
+
+        return int(self.__config.maximum_runtime_milliseconds) \
+            < remaining_time_in_millis
+
+    def process_messages(self):
+        """
+        Iterates over and processes messages in a MESH inbox
+        """
+        is_message_iterator_empty = False
+
+        while not is_message_iterator_empty:
+            self.__log.info('Polling for messages')
+
+            # if iterate_all_messages does not return any items, we will exit the loop
+            is_message_iterator_empty = True
+
+            # Initial processing of each message
+            for message in self.__mesh_client.iterate_all_messages():
+                is_message_iterator_empty = False
+                if not self.is_enough_time_to_process_message():
+                    self.__log.info(
+                        'Not enough time to process more files. Exiting')
+                    self.__polling_metric.record(1)
+                    return
+
+                self.process_message(message)
+
+        self.__log.info('No new messages found. Exiting')
+        self.__polling_metric.record(1)
+
+    def process_message(self, message):
+        """
+        Processes an individual message from a MESH inbox - validates sender and publishes event
+        """
+
+        message_type = getattr(message, 'message_type', '')
+        sender_mailbox_id = getattr(message, "sender", "")
+        workflow_id = getattr(message, "workflow_id", "")
+        subject = getattr(message, "subject", "")
+        message_reference = getattr(message, "local_id", "")
+
+        logger = self.__log.bind(
+            message_id=message.id(),
+            sender=sender_mailbox_id,
+            workflow_id=workflow_id,
+            subject=subject,
+            local_id=message_reference,
+            message_type=message_type,
+        )
+
+        logger.info(PROCESSING_MESSAGE)
+
+        try:
+            # Basic sender validation - only publish events for known senders
+            if not self.__sender_lookup.is_valid_sender(sender_mailbox_id):
+                raise AuthorizationError(
+                    f'Cannot authorize sender with mailbox ID "{sender_mailbox_id}"')
+
+            # Get the corresponding sender ID
+            sender_id = self.__sender_lookup.get_sender_id(sender_mailbox_id)
+
+            # Publish event for valid sender
+            message_id = message.id()
+            event_detail = {
+                "data": {
+                    "meshMessageId": message_id,
+                    "senderId": sender_id,
+                    "messageReference": message_reference
```

Comment:
I can be wrong, but I thought we were going to "namespace" the message_reference with the sender_id or mailboxid?